tools/file_packager.py

```diff
@@ -993,7 +993,9 @@ def generate_js(data_target, data_files, metadata):
 
             const chunks = [];
             const headers = response.headers;
-            const total = Number(headers.get('Content-Length') ?? packageSize);
+            const contentEncoding = headers.get('Content-Encoding');
+            const isCompressed = contentEncoding === 'gzip' || contentEncoding === 'br'; 
+            const total = Number(isCompressed ? packageSize ?? headers.get('Content-Length') : headers.get('Content-Length') ?? packageSize);
```

Comment:
The combination of nullish coalescing and ternary operator makes this pretty hard to read.

How about this instead (assuming this is what you are trying to express.

```
const total = packageSize;
// Ignore Content-Length when compression is used since that reports the compressed size
if (!isCompressed && headers.get('Content-Length')) {
  total = Number(headers.get('Content-Length'));
}
``

Reply:
@sbc100 Yes close, may be below when packageSize is null

```
const total = packageSize;
if (!total || (!isCompressed && headers.get('Content-Length'))) {
  total = Number(headers.get('Content-Length'));
}

```

Reply:
Can you re-write in that form ?

Reply:
Done, Thanks